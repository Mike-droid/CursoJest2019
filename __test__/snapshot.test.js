import {getCharacter} from './../snapshot';
import rick from './../rick.json';

describe('Es hora de las instantaneas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  //*Se crearán datos aletorios que no van a coincidir con el snapshot
  test('Siempre fallará la instantanea', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    }
    expect(user).toMatchSnapshot();
  });

  test('Tenemos una excepción dentro del código', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Miguel Reyes"
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number) //*ID tendrá cualquier valor numérico
    });
  });
});