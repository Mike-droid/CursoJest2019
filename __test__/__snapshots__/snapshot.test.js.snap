// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallará la instantanea 1`] = `
Object {
  "createAt": 2021-03-31T16:43:22.860Z,
  "id": 6,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepción dentro del código 1`] = `
Object {
  "id": Any<Number>,
  "name": "Miguel Reyes",
}
`;
